pkg/cableengine/ipsec/strongswan_test.go

```diff
@@ -0,0 +1,68 @@
+package ipsec
+
+import (
+	"bytes"
+	"os"
+	"testing"
+
+	"github.com/kelseyhightower/envconfig"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+)
+
+var _ = Describe("Util", func() {
+	Describe("Charon port configuration", testCharonConfigPortsGen)
+})
+
+func testCharonConfigPortsGen() {
+	Context("config rendering", func() {
+		It("should render config correctly out of strongSwan parameters", func() {
+			ss := strongSwan{ipSecIKEPort: "500", ipSecNATTPort: "4500"}
+			buf := new(bytes.Buffer)
+
+			ss.renderCharonConfigTemplate(buf)
+
+			Expect(buf.String()).To(ContainSubstring("port = 500"))
+			Expect(buf.String()).To(ContainSubstring("port_nat_t = 4500"))
+			Expect(buf.String()).To(ContainSubstring("make_before_break = yes"))
+			Expect(buf.String()).To(ContainSubstring("ignore_acquire_ts = yes"))
+
+		})
+	})
+
+	Context("environment variable processing", func() {
+		It("should get the defaults from the specification definition", func() {
+			checkEnvVarParsingPorts("500", "4500")
+		})
+
+		It("should get the rigth environment variable names", func() {
+			const (
+				ikePort        = "555"
+				nattPort       = "4555"
+				ikePortEnvVar  = "CE_IPSEC_IKEPORT"
+				nattPortEnvVar = "CE_IPSEC_NATTPORT"
+			)
+			os.Setenv(ikePortEnvVar, ikePort)
+			os.Setenv(nattPortEnvVar, nattPort)
+
+			checkEnvVarParsingPorts(ikePort, nattPort)
+
+			os.Unsetenv(ikePortEnvVar)
+			os.Unsetenv(nattPortEnvVar)
+		})
+	})
+}
+
+func checkEnvVarParsingPorts(ikePort string, nattPort string) {
+	ipSecSpec := specification{}
+	err := envconfig.Process(ipsecSpecEnvVarPrefix, &ipSecSpec)
+	Expect(err).NotTo(HaveOccurred())
+	Expect(ipSecSpec.IKEPort).To(Equal(ikePort))
+	Expect(ipSecSpec.NATTPort).To(Equal(nattPort))
```

Comment:
This is really just testing the external `envconfig.Process` lib function.  It seems it would be more useful to test `NewStrongSwan` and verify the returned `ipSecIKEPort`, `ipSecNATTPort ` et al are set properly?

Reply:
Yeah, I understand what you mean, and I thought about it, but let me share my line of thinking:

I wasn't sure about how envconfig.Process generated the env var names, and I wanted to test End to End, and produce self-documenting test about what goes in (which env-vars) and what do you get in the end (Assuming that envconfig.Process will always work)

Does that sound ok?

Reply:
ok so you wrote the test originally b/c you weren't sure exactly how envconfig.Process worked - that's makes sense. But since this is mainly what `NewStrongSwan` does internally, do you think it makes sense now to test `NewStrongSwan` to get a bit more coverage and e2e (and make it more blackbox)?

Reply:
Let's merge this as is. I can push a PR later to propose the change.